wonk/models.py

```diff
@@ -153,10 +176,40 @@ def canonicalize_actions(actions: Set[str]) -> Union[str, List[str]]:
 def canonicalize_resources(resources: Set[str]) -> Union[str, List[str]]:
     """Return the set of resources as either a single string or a sorted list of strings."""
 
+    if "*" in resources:
+        return "*"
+
     if len(resources) == 1:
         return next(iter(resources))
 
-    return sorted(resources)
+    result = []
+    last_wildcard = None
+    for resource in sorted(resources):
+        if last_wildcard is not None:
+            # Skip this resource if it matches the last wildcard we saw.
+            if resource.startswith(last_wildcard):
+                continue
```

Comment:
This approach is fairly different than the one above for actions. Can we either add some documentation around the reason for that or use the same approach for both?

Reply:
Good point. I refactored those functions to use the same core algorithm.